code/modules/projectiles/projectile.dm

```diff
@@ -524,9 +349,299 @@
 		SP.SetName((name != "shrapnel")? "[name] shrapnel" : "shrapnel")
 		SP.desc += " It looks like it was fired from [shot_from]."
 		return SP
-	
+
 /obj/item/projectile/get_autopsy_descriptors()
 	return list(name)
 
 /obj/item/projectile/Process_Spacemove()
 	return TRUE	//Bullets don't drift in space
+
+/obj/item/projectile/proc/old_style_target(atom/target, atom/source)
+	if(!source)
+		source = get_turf(src)
+	setAngle(Get_Angle(source, target))
+
+/obj/item/projectile/proc/fire(angle, atom/direct_target)
+	//If no angle needs to resolve it from xo/yo!
+	if(direct_target)
+		direct_target.bullet_act(src, def_zone)
+		on_impact(direct_target)
+		qdel(src)
+		return
+	if(isnum(angle))
+		setAngle(angle)
+	// trajectory dispersion
+	var/turf/starting = get_turf(src)
+	if(!starting)
+		return
+	if(isnull(Angle))	//Try to resolve through offsets if there's no angle set.
+		if(isnull(xo) || isnull(yo))
+			crash_with("WARNING: Projectile [type] deleted due to being unable to resolve a target after angle was null!")
+			qdel(src)
+			return
+		var/turf/target = locate(Clamp(starting + xo, 1, world.maxx), Clamp(starting + yo, 1, world.maxy), starting.z)
+		setAngle(Get_Angle(src, target))
+	if(dispersion)
+		var/DeviationAngle = (dispersion * 15)
+		setAngle(Angle + rand(-DeviationAngle, DeviationAngle))
+	original_angle = Angle
+	if(!nondirectional_sprite)
+		var/matrix/M = new
+		M.Turn(Angle)
+		transform = M
+	forceMove(starting)
+	trajectory = new(starting.x, starting.y, starting.z, 0, 0, Angle, pixel_speed)
+	last_projectile_move = world.time
+	fired = TRUE
+	if(hitscan)
+		return process_hitscan()
+	else
+		if(!is_processing)
+			START_PROCESSING(SSprojectiles, src)
+		pixel_move(1)	//move it now!
+
+/obj/item/projectile/proc/preparePixelProjectile(atom/target, atom/source, params, angle_offset = 0)
+	var/turf/curloc = get_turf(source)
+	var/turf/targloc = get_turf(target)
+	forceMove(get_turf(source))
+	starting = get_turf(source)
+	original = target
+
+	var/list/calculated = list(null,null,null)
+	if(isliving(source) && params)
+		calculated = calculate_projectile_angle_and_pixel_offsets(source, params)
+		p_x = calculated[2]
+		p_y = calculated[3]
+		setAngle(calculated[1])
+
+	else if(targloc && curloc)
+		yo = targloc.y - curloc.y
+		xo = targloc.x - curloc.x
+		setAngle(Get_Angle(src, targloc))
+	else
+		crash_with("WARNING: Projectile [type] fired without either mouse parameters, or a target atom to aim at!")
+		qdel(src)
+	if(angle_offset)
+		setAngle(Angle + angle_offset)
+
+/obj/item/projectile/Crossed(atom/movable/AM) //A mob moving on a tile with a projectile is hit by it.
+	..()
+	if(isliving(AM) && (AM.density || AM == original) && !(pass_flags & PASS_FLAG_MOB))
+		Bump(AM)
+
+/obj/item/projectile/Initialize()
```

Comment:
Just define `permutated` as `= list()`, having a proc just to initialize it is pointless.

Reply:
It was already defined as a list, oddly enough. This was a direct port, so assume auroracode oddities.